### DISCUSSION TITLE
.NET 8 Performance Edition

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#54] [General]
# .NET 8 Performance Edition

As with every .NET release, Microsoft improves the performance of the runtime and guess what: This release is no exception to this. In this blog post, I want to go through some of the improvements made so far (.NET 8 preview 3).

https://steven-giesel.com/blogPost/f6504300-7bf0-48d2-8a14-ba4e2bbea02e

**Comment by ghadzhigeorgiev** [comment]
All speed improvements are welcome!

**Comment by MiheevN** [comment]
You need to strive to keep up with compiled languages.  And ideally surpass.

**Reply by wekempf** [reply]
C# is a compiled language.

**Reply by MiheevN** [reply]
Only now is becoming, with AOT.  And before that, it requires runtime.  The compiled language is C, C++, Rust, and TD.

**Reply by linkdotnet** [reply]
C# ,much like java, is considered a compilable language.

**Reply by MiheevN** [reply]
Then what is the correct way to name languages ​​without Runtime?  which I have listed.  they are faster.  And we have to include libraries, and spend time on Invoke

**Reply by linkdotnet** [reply]
Golang also includes libraries into the executable.I am not sure if there is a hard line or other word for languages without a runtime - except that.

**Reply by fabio-s-franco** [reply]
There are several ways to distinguish both flavours of compiled languages, or more precisely, binaries.
Jitted (from Just-In-Time native code compilation) vs native code.
Managed vs unmanaged.
Bytecode vs machine code compiled.

And the assumption that machine code compiled languages are faster is not true, not all the time anyways. Remember that optimisations can vary wildly between two contenders and in cenarios where same code is executed many times over, the initial overhead of JIT code compilation is quickly diluted and eventually boils down to which is better optimised. With machine code compiled languages, a lot of these optimisations are responsibility of the programmer writing it and that alone can be a big gotcha when comparing performances of non-trivial implementations.

**Comment by mizrael** [comment]
very nice article, thanks! 
It would be nice to have:
- comparison with .NET 6 (being LTE)
- benchmarks with common collection operations (add, lookup, removal)

**Reply by linkdotnet** [reply]
That is a very nice idea! I will substitute the article later today with the results of net60 and some list operations!

**Reply by linkdotnet** [reply]
Updated the article with the latest benchmarks! Credits to you @mizrael - thanks

**Reply by mizrael** [reply]
thanks! The benchmark results for List don't show up though.

Also, I don't want to be pedantic but probably 1000 entries are not enough to get real value from those tests.
Moreover, lookups in Lists are really inefficient, can you use Dictionaries or HashSets instead?

**Reply by linkdotnet** [reply]
I updated the post - i just forgot to update that section. Yeah I can use dictionaries for lookups

**Reply by linkdotnet** [reply]
Updated it again - hope those benchmarks make more sense (10k elements)

**Reply by mizrael** [reply]
thanks!

**Comment by AmSmart** [comment]
Nice article!
